### PR TITLE
Added new widgets: YMenuBar, YItemSelector

### PR DESCRIPTION
**Pull request by shundhammer**
This adds the new widgets that were added to libyui in recent months to the bindings:

- YMenuBar https://github.com/libyui/libyui/pull/169
- YItemSelector  https://github.com/libyui/libyui/pull/152

This was requested by @anaselli so they can finally migrate from the MGAMenuBar to this new common YMenuBar.

Please notice that I don't know much about SWIG bindings; I just added lines that were similar to existing ones (YMenuButton).

Most of YMenuButton is now in a new base class YMenuWidget that is also inherited by YMenuBar, so both widgets share the parts that might be sensitive (`addItem( YWidget *item_disown )`).

YItemSelector is another YSelectionWidget that also heavily relies on its base class; I don't expect any trouble there.

---------------------

@dmulder : I was told you are interested in this package from the SUSE side. Can you have a look at this PR?

### CODE REVIEW

**Comment by dmulder** [comment]
@shundhammer looks ok to me.

**Comment by shundhammer** [comment]
@dmulder : Thanks!

File: swig/yui.i [hunk 1]
```diff
@@ -61,10 +61,12 @@ SWIGEXPORT void Init__yui(void) {
 #include "yui/YImage.h"
 #include "yui/YInputField.h"
 #include "yui/YItem.h"
+#include "yui/YItemSelector.h"
 #include "yui/YLogView.h"
 #include "yui/YMacro.h"
 #include "yui/YMacroPlayer.h"
 #include "yui/YMacroRecorder.h"
+#include "yui/YMenuBar.h"
```
Comment (anaselli) [inline swig/yui.i:69]: YMenuWidget is missing, please add also
#include "yui/YMenuWidget.h"

File: swig/yui.i [hunk 2]
```diff
@@ -214,12 +216,14 @@ class Exception;
...
 %include yui/YInputField.h
 %include yui/YIntField.h
+%include yui/YItemSelector.h
 %include yui/YLabel.h
 %include yui/YLayoutBox.h
 %include yui/YLogView.h
 %include yui/YMacro.h
 %include yui/YMacroPlayer.h
 %include yui/YMacroRecorder.h
+%include yui/YMenuBar.h
```
Comment (anaselli) [inline swig/yui.i:226]: YMenuWidget is missing, please add also
%include yui/YMenuWidget.h

**Review by anaselli — COMMENTED** [review]
You could also add this [python example](https://gist.github.com/anaselli/af9597ae0a81bc564bdb3de2e56e78d9) under swig/python/examples

File: swig/yui.i (hunk 2 above)
Comment (shundhammer) [reply swig/yui.i:227]: It doesn't support indirect includes? Eeek...

Adding.

**Comment by mvidner** [comment]
I have a build fix for the Travis failure...

**Comment by mvidner** [comment]
@shundhammer please cherry pick https://github.com/libyui/libyui-bindings/commit/2abc86f8f27523b75b2f8465b0a7422aea3f377f 

(and if you in the future make PRs from a branch in the main repo instead of from your fork, I will be able to simply add the commit instead of bothering you to do it)

**Comment by mvidner** [comment]
@anaselli I have added your example, but it has not worked for me until I applied the fix below. The runtime error was

> self.menubar.resolveShortcutConflicts()
> AttributeError: 'YMenuBar' object has no attribute 'resolveShortcutConflicts'

and swig 4.0.2 was saying at compile time 
> /usr/include/yui/YMenuBar.h:50: Warning 401: Base class 'YMenuWidget' undefined.
/usr/include/yui/YMenuWidget.h:39: Warning 401: 'YMenuWidget' must be defined before it is used as a base class.

I wonder if you've seen this.

Fix:
```diff
diff --git a/swig/yui.i b/swig/yui.i
index b90a336..55b3878 100644
--- a/swig/yui.i
+++ b/swig/yui.i
@@ -66,10 +66,10 @@ SWIGEXPORT void Init__yui(void) {
 #include "yui/YMacro.h"
 #include "yui/YMacroPlayer.h"
 #include "yui/YMacroRecorder.h"
+#include "yui/YMenuWidget.h"
 #include "yui/YMenuBar.h"
 #include "yui/YMenuButton.h"
 #include "yui/YMenuItem.h"
-#include "yui/YMenuWidget.h"
 #include "yui/YMultiLineEdit.h"
 #include "yui/YMultiProgressMeter.h"
 #include "yui/YMultiSelectionBox.h"
@@ -224,10 +224,10 @@ class Exception;
 %include yui/YMacro.h
 %include yui/YMacroPlayer.h
 %include yui/YMacroRecorder.h
+%include yui/YMenuWidget.h
 %include yui/YMenuBar.h
 %include yui/YMenuButton.h
 %include yui/YMenuItem.h
-%include yui/YMenuWidget.h
 %include yui/YMultiLineEdit.h
 %include yui/YMultiProgressMeter.h
 %include yui/YMultiSelectionBox.h
```

**Comment by anaselli** [comment]
@mvidner yes i wrote for those missing entries in my review comments...

**Review by mvidner — APPROVED** [review]

**Pull request merged**